overlord/snapstate/handlers.go

```diff
@@ -241,14 +245,17 @@ func (m *SnapManager) installOneBaseOrRequired(st *state.State, snapName, channe
 	return ts, err
 }
 
-func (m *SnapManager) installPrereqs(t *state.Task, base string, prereq []string, userID int) error {
+func (m *SnapManager) installPrereqs(t *state.Task, base string, prereq []string, userID int, tm timings.Measurer) error {
 	st := t.State()
 
 	// We try to install all wanted snaps. If one snap cannot be installed
 	// because of change conflicts or similar we retry. Only if all snaps
 	// can be installed together we add the tasks to the change.
 	var tss []*state.TaskSet
 	for _, prereqName := range prereq {
+		nestedTiming := tm.StartSpan("install-prereq", fmt.Sprintf("Install %q", prereqName))
+		defer nestedTiming.Stop()
```

Comment:
defer are executed at the end of the function, not the blockc so they cannot be used for this in a loop

Reply:
Thanks for catching, will be fixed in the separate PR, per comment above.